### PR TITLE
Support Select Lists

### PR DESCRIPTION
**Pull request by smcleodau**
When adding attachment options, you can now include Select Lists, not just buttons. I added an enum to support the type of Select option and attribute converters to make creating Attachment Options easier.

### CODE REVIEW

File: SlackAPI/SlackAPI.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
```
Comment (Inumedia) [inline SlackAPI/SlackAPI.csproj]: @gpailler , Correct me if I'm wrong, but updating the Newtonsoft version here would require anyone using the library to also have to update their Newtonsoft to `>=11.0.2`?

File: SlackAPI/SlackClient.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@
 namespace SlackAPI
 {
     /// <summary>
-    /// SlackClient is intended to solely handle RPC (HTTP-based) functionality. Does not handle WebSocket connectivity.
+    /// SlackClient is intendedfty. Does not handle WebSocket connectivity.
```
Comment (Inumedia) [inline SlackAPI/SlackClient.cs]: ??? This shouldn't have been changed

File: SlackAPI/SlackClient.cs [hunk 3]
```diff
@@ -493,7 +493,7 @@ public void EmitLogin(Action<LoginResponse> callback, string agent = "Inumedia.S
             if (attachments != null && attachments.Length > 0)
                 parameters.Add(new Tuple<string, string>("attachments", JsonConvert.SerializeObject(attachments)));
 
-            parameters.Add(new Tuple<string, string>("as_user", as_user.ToString()));
+                parameters.Add(new Tuple<string, string>("as_user", as_user.ToString()));
```
Comment (Inumedia) [inline SlackAPI/SlackClient.cs]: Why is the indentation here changed?

File: SlackAPI/SlackClientBase.cs [hunk 4]
```diff
@@ -110,7 +110,7 @@ protected HttpResponseMessage PostRequest(string requestUri, MultipartFormDataCo
             return httpClient.PostAsync(requestUri, form).Result;
         }
 
-        public void RegisterConverter(JsonConverter converter)
+        public void RegisterConverter(Newtonsoft.Json.JsonConverter converter)
```
Comment (Inumedia) [inline SlackAPI/SlackClientBase.cs]: Why was this changed to `Newtonsoft.Json.JsonConverter`? It has the using at the top of the file?

File: SlackAPI/SlackTaskClient.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
 ﻿using Newtonsoft.Json;
+using SlackAPI.Utilities;
```
Comment (Inumedia) [inline SlackAPI/SlackTaskClient.cs]: Why was this added here?

File: SlackAPI/AttachmentActionOption.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+{
+    //See: https://api.slack.com/docs/message-menus
+    public class AttachmentActionOption
+    {
+        public AttachmentActionOption(string text, string value)
+        {
+            this.text = text;
+            this.value = value;
+        }
+        public string text { get; }
```
Comment (Inumedia) [inline SlackAPI/AttachmentActionOption.cs]: Why is this a property but the other isn't?

File: SlackAPI/AttachmentActionOption.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+﻿namespace SlackAPI
+{
+    //See: https://api.slack.com/docs/message-menus
```
Comment (Inumedia) [inline SlackAPI/AttachmentActionOption.cs]: This should be a proper XML document so people using the library can see what to reference

File: SlackAPI/Attachment.cs [hunk 8]
```diff
@@ -1,4 +1,6 @@
-﻿namespace SlackAPI
+﻿using System.Collections.Generic;
```
Comment (Inumedia) [inline SlackAPI/Attachment.cs]: Why was this using added but no code was changed that would indicate this is needed?

File: SlackAPI/SlackAPI.csproj (hunk 1 above)
Comment (gpailler) [reply SlackAPI/SlackAPI.csproj]: From what I know yes. It's better to target the lowest version to maximize the compatibility.

**Comment by Inumedia** [comment]
@smcleodau The review from before still sticks, these things need to be changed before this can be merged in

**Comment by Inumedia** [comment]
@smcleodau I know I took 3 months to respond to the PR initially, but if I don't hear back or see changes before end of June, I'm going to close this PR

**Comment by smcleodau** [comment]
That's fine, go ahead. We've managed to move on and all is good.
